src/components/formControl/index.tsx

```diff
@@ -0,0 +1,50 @@
+import React, { FC, PropsWithChildren } from 'react';
+import {
+  FormControl as ChakraFormControl,
+  FormErrorMessage,
+  FormHelperText,
+  FormLabel,
+} from '@chakra-ui/react';
+
+type Props = {
+  isDisabled?: boolean;
+  isRequired?: boolean;
+  errorMessage?: string;
+  id: string;
+  label?: string;
+  hint?: string;
+};
+
+const FormControl: FC<PropsWithChildren<Props>> = ({
+  children,
+  isDisabled,
+  isRequired,
+  errorMessage,
+  id,
+  label,
+  hint,
+}) => {
+  const isInvalid = !!errorMessage;
+
+  return (
+    <ChakraFormControl
+      isDisabled={isDisabled}
+      isInvalid={isInvalid}
+      isRequired={isRequired}
+      id={id}
+    >
+      {label ? (
+        <FormLabel htmlFor={id}>
+          {label}
+          {isRequired ? <>&nbsp;</> : null}
+        </FormLabel>
+      ) : null}
+      {children}
+      {hint && !isInvalid ? <FormHelperText>{hint}</FormHelperText> : null}
+      <FormErrorMessage>{errorMessage}</FormErrorMessage>
```

Comment:
If there is no `errorMessage` are we still going to have `<FormErrorMessage>` div in the DOM or is that automatically handled by `<ChakraFormControl>`? Or should we do something like this 
`{ isInvalid && <FormErrorMessage>{errorMessage}</FormErrorMessage> }`

Reply:
> [docs](https://chakra-ui.com/docs/components/form-control/usage#error-message) FormErrorMessage will only show up when the property isInvalid in FormControl is true.

Reply:
Essentially chakra's FormControl provides a context that the children use

Reply:
That's what I thought. Read the docs then tested it in the playground and had some strange behavior so wasn't sure anymore. Now I tested it again on fresh playground in incognito and it works. Thanks for explanation!